src/Data/Parsers/ResourceIdSplitter.test.ts

```diff
@@ -0,0 +1,265 @@
+      const resourceId = "my_resource::my_type[agent,key=value]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "my_resource::my_type",
+        agent: "agent",
+        value: "value",
+      });
+    });
+
+    it("should parse resource ID with complex agent name", () => {
+      const resourceId = "resource::type[agent-with-dashes,key=value]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent-with-dashes",
+        value: "value",
+      });
+    });
+
+    it("should parse resource ID with complex value containing special characters", () => {
+      const resourceId = "resource::type[agent,key=value-with-special-chars_123]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent",
+        value: "value-with-special-chars_123",
+      });
+    });
+
+    it("should throw error for resource ID with empty value", () => {
+      const resourceId = "resource::type[agent,key=]";
+
+      expect(() => splitResourceId(resourceId)).toThrow(
+        "Invalid resource id: resource::type[agent,key=]"
+      );
+    });
+
+    it("should parse resource ID with numeric value", () => {
+      const resourceId = "resource::type[agent,port=8080]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent",
+        value: "8080",
+      });
+    });
+
+    it("should parse resource ID with UUID as value", () => {
+      const resourceId = "resource::type[agent,id=550e8400-e29b-41d4-a716-446655440000]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent",
+        value: "550e8400-e29b-41d4-a716-446655440000",
+      });
+    });
+
+    it("should parse resource ID with complex key-value pair", () => {
+      const resourceId = "resource::type[agent,complex_key=complex_value_with_underscores]";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent",
+        value: "complex_value_with_underscores",
+      });
+    });
+  });
+
+  describe("invalid resource IDs", () => {
+    it("should throw error for resource ID without brackets", () => {
+      const resourceId = "resource::type";
+
+      expect(() => splitResourceId(resourceId)).toThrow("Invalid resource id: resource::type");
+    });
+
+    it("should parse resource ID with incomplete brackets (missing closing bracket)", () => {
+      const resourceId = "resource::type[agent,key=value";
+      const result = splitResourceId(resourceId);
+
+      expect(result).toEqual({
+        type: "resource::type",
+        agent: "agent",
+        value: "value",
+      });
+    });
```

Comment:
EDIT: ignore this comment 
> 
> Is there a reason why we want this behaviour ? I don't have a good feel for how realistic it is/what the flow is like, but I feel this could lead to errors e.g.:
> - full resource id is `resource::type[agent,key=value_untruncated]`
> - for some reason it gets truncated to `resource::type[agent,key=value`
> - `splitResourceId` will silently parse `value=value` instead of `value=value_untruncated`
>

Reply:
Ignore above, just realized this parsing was a temporary work around https://github.com/inmanta/web-console/issues/6565